### PR TITLE
Make JDQZPP optional

### PR DESCRIPTION
**Pull request by Sbte**
I made JDQZPP optional through the Continuation, meaning I now also had to validate the JDQZPP parameters. I had to make some changes for that as well, so please check those.

### CODE REVIEW

**Comment by Sbte** [comment]
You have to update JDQZPP

File: src/tests/test_utils.C [hunk 1]
```diff
@@ -170,6 +170,9 @@ checkParameterListAgainstDefaultAndOverrides
         parameters.insert(pair.first);
     }
 
+    Teuchos::ParameterList defaultedList;
+    defaultedList.validateParametersAndSetDefaults(defaultList);
+
```
Comment (merijn) [inline src/tests/test_utils.C:175]: I don't really understand why this change, but I assume there's a reason? Other than that I don't think there's anything weird in this commit.

Comment (Sbte) [reply src/tests/test_utils.C:175]: It makes all the parameters in `defaultedList` `[default]`. Since we use `set` in JDQZPP, and later everywhere in I-EMIC (when we start adding documentation), the values in the list that is returned from `getDefaultParameters` (`defaultList`) are not `[default]`. However, we expect the list that is being compared against to be all `[default]`, so we need these values also to be `[default]` to make them equal.

Comment (merijn) [reply src/tests/test_utils.C:175]: Ah, right. Rest looks fine to me.
